<?xml version="1.0" encoding="UTF-8"?>
<launch>

  <arg name="kinova_robotType" default="j2s7s300"/>
  <arg name="kinova_robotName" default="$(arg kinova_robotType)"/>


  <!-- The name of the parameter under which the URDF is loaded -->
  <param name="arm_robot_description"
    command="$(find xacro)/xacro --inorder '$(find kinova_description)/urdf/$(arg kinova_robotType)_standalone.xacro'" />
  <arg name="arm_robot_description" default="arm_robot_description"/>
  <!-- The semantic description that corresponds to the URDF -->
  <param name="$(arg arm_robot_description)_semantic" textfile="$(find j2s7s300_moveit_config)/config/j2s7s300.srdf" />

  <node pkg="tf" type="static_transform_publisher" name = "mpc_base2movo_left_arm_base" args="0 0 -0.09425 0 0 0   /left_base_link /left_mpc_base_link 10" />
  <node pkg="tf" type="static_transform_publisher" name = "mpc_ee2movo_left_arm_ee" args="0 0 0 0.5 -0.5 0.5 -0.5  /left_ee_link /left_mpc_ee_link 10" />
  <node pkg="tf" type="static_transform_publisher" name = "mpc_base2movo_right_arm_base" args="0 0 -0.09425 0 0 0   /right_base_link /right_mpc_base_link 10" />
  <node pkg="tf" type="static_transform_publisher" name = "mpc_ee2movo_right_arm_ee" args="0 0 0 0.5 -0.5 0.5 -0.5  /right_ee_link /right_mpc_ee_link 10" />

  <!--  rviz visualization-->
  <arg name="command_args" value="-d $(find mpc)/config/real_robot_rviz_vis.rviz" />
  

  <rosparam file="$(find mpc)/config/nlmpc.yaml" command="load" /> 
  <node name="left_movo_arm_node" pkg="mpc" type="main_movo_sc" output="screen">
    <param name="arm_name_space" type="string" value="left" />
    <param name="plot_obstacle" type="bool" value="false" />
    <remap from="robot_description" to="arm_robot_description"/>
  </node>
  <node name="right_movo_arm_node" pkg="mpc" type="main_movo_sc" output="screen">
    <param name="arm_name_space" type="string" value="right" />
    <param name="plot_obstacle" type="bool" value="true" />
    <remap from="robot_description" to="arm_robot_description"/>
  </node> 
</launch>
